lib/Raven/Serializer.php

```diff
@@ -67,17 +67,44 @@ public function __construct($mb_detect_order = null)
      * sanitization and encoding.
      *
      * @param mixed $value
+     * @param array $context
+     * @return string|bool|double|int|null|object|array
+     */
+    public function serialize($value, array $context = [])
+    {
+        $full_context = $this->get_full_context($context);
+        return $this->serialize_inner($value, $full_context['max_depth'], 0);
+    }
+
+    /**
+     * @param array $context
+     * @return array
+     */
+    protected function get_full_context(array $context)
+    {
+        if (!array_key_exists('max_depth', $context)) {
```

Comment:
The `max_depth` option (correct me if I'm wrong) seems to be always set from the lines of code that call the serialize function, so backwards compatibility seems to be guaranteed. With these lines of code, the user won't be able to serialize everything regardless of depth because a max depth of 3 is always forced to be set